packages/flutter/lib/src/material/refresh_indicator.dart

```diff
@@ -261,8 +261,14 @@ class RefreshIndicatorState extends State<RefreshIndicator> with TickerProviderS
     super.dispose();
   }
 
+  // The [dragDetails] is non-null if the scroll notifications are triggered by dragging.
+  // The indicator can only be triggered by dragging, and cannot be triggered
+  // by other non-dragging behaviors such as [ScrollController.jumpTo].
+  // See also [RefreshIndicator.triggerMode].
   bool _shouldStart(ScrollNotification notification) {
-    return (notification is ScrollStartNotification || (notification is ScrollUpdateNotification && notification.dragDetails != null && widget.triggerMode == RefreshIndicatorTriggerMode.anywhere))
+    return ((notification is ScrollStartNotification && notification.dragDetails != null) // Start dragging at zero scroll position.
```

Comment:
now i looked at this again lets break up this function a bit to be more

```dart
bool _shouldStart(ScrollNotification notification) {
    // If the notification.dragDetails is null, this scroll is not triggered by user dragging. It may be a result of ScrollController.jumpTo or a ballistic scroll. In this case we don't want to trigger the refresh indicator.
    if (notification.dragDetails == null)
      return false;
    return (notification is ScrollStartNotification || (notification is ScrollUpdateNotification && 
   widget.triggerMode == RefreshIndicatorTriggerMode.anywhere))
      && notification.metrics.extentBefore == 0.0
      && _mode == null
      && _start(notification.metrics.axisDirection);
  }
```

Reply:
It seems that the logic is much clearer. 👍

Reply:
Considering the performance, I had to change it to what it was before, but modify the comment as you suggested.
```dart
  bool _shouldStart(ScrollNotification notification) {
    // If the notification.dragDetails is null, this scroll is not triggered by
    // user dragging. It may be a result of ScrollController.jumpTo or a ballistic scroll.
    // In this case we don't want to trigger the refresh indicator.
    return ((notification is ScrollStartNotification && notification.dragDetails != null)
            || (notification is ScrollUpdateNotification && notification.dragDetails != null && widget.triggerMode == RefreshIndicatorTriggerMode.anywhere))
      && notification.metrics.extentBefore == 0.0
      && _mode == null
      && _start(notification.metrics.axisDirection);
  }
```
Any suggestions?